core/src/main/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsRequest.java

```diff
@@ -154,7 +133,9 @@ public void readFrom(StreamInput in) throws IOException {
         routing = in.readOptionalString();
         preference = in.readOptionalString();
 
-        types = in.readStringArray();
+        if (in.getVersion().onOrBefore(Version.V_6_0_0_alpha1)) {
+            in.readStringArray();
```

Comment:
maybe leave a quick comment, eg. `// types` so that the reader can know what this parameter was about